tofu/geom/_core.py

```diff
@@ -1916,50 +1936,73 @@ def from_txt(
                 - Hence the number of lines hould be N = npts + noccur + 1
                 - Lines 1:npts+1 contain the polygon points
                 - Lines npts+1: contain positions and extent of each occurence
-        out:    str
+        returnas:    str
             Flag indicating whether to return:
                - 'dict'  : a dictionnary of np.ndarrays
                - 'object': a tofu.geom.Struct subclass, using the other kwdargs
+        warn:       None / bool
+            Whether to raise a warning if the formatting of the file is
+            suspicious
 
         Return
         ------
         obj:    tf.geom.Struct sublass instance  / dict
-            Depending on the value of out, obj can be:
+            Depending on the value of returnas, obj can be:
                 - An instance of the relevant tofu.geom.Struct subclass
                 - A dict with keys 'poly', 'pos' and 'extent'
         """
 
-        if out not in [object, "object", "dict"]:
-            msg = "Arg out must be either:"
-            msg += "    - 'object': return a %s instance\n" % cls.__name__
-            msg += "    - 'dict' : return a dict with polygon, pos and extent"
+        # Check inputs
+        if returnas not in [object, 'object', dict, 'dict']:
+            msg = ("Arg returnas must be either:"
+                   + "\t- 'object': return {} instance\n".format(cls.__name__)
+                   + "\t- 'dict' : return a dict with polygon, pos and extent")
             raise Exception(msg)
-        if not pfe[-4:] == ".txt":
-            msg = "Only accepts .txt files (fed to np.loadtxt) !\n"
-            msg += "    file:  %s" % pfe
+        if pfe[-4:] not in ['.txt', '.csv']:
```

Comment:
I read recentley that a better way to check for file extensions is to do 
``pfe.lower().endswith(('.txt', '.csv'))``
maybe the lower part is unnecessary. But it would alow to check something like:
``pfe.lower().endswith(('.txt', '.csv', '.jpeg', '.m'))``
so no need to cound the number of characters and is more compact and versatile

Reply:
Very good advice, thanks, I'll correct!

Reply:
Good to hear from you !  :-)